openedx/core/djangoapps/credit/signature.py

```diff
@@ -16,17 +16,30 @@
 
 """
 
+import logging
 import hashlib
 import hmac
 
 from django.conf import settings
 
 
+log = logging.getLogger(__name__)
+
+
 def get_shared_secret_key(provider_id):
     """
     Retrieve the shared secret key for a particular credit provider.
     """
-    return getattr(settings, "CREDIT_PROVIDER_SECRET_KEYS", {}).get(provider_id)
+    secret = getattr(settings, "CREDIT_PROVIDER_SECRET_KEYS", {}).get(provider_id)
+
+    if isinstance(secret, unicode):
+        try:
+            secret = str(secret)
```

Comment:
Good call, I missed the bit about the spec requiring a 32-character ASCII string. In that case, I might be even more explicit and use `secret.encode('ascii')`, but that's a minor.